scripts/bump_major_version.py

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import argparse
+import shutil
+import yaml
+
+# With each new major release of CockroachDB, there are a number of
+# administrative tasks that need to be accomplished on the docs website.  This
+# includes, but is not limited to:
+
+# - Create a new `vXY.Z` folder, and copy content to it
+#
+# - Copy includes to a new `_includes/vXY.Z` folder
+#
+# - Copy images to a new versioned folder, and edit image link text in the
+# referencing documents as needed to point to the right places
+#
+# - Update the YAML config files with the new version info
+#
+# - Create a new sidebar JSON file
+#
+# - In the newly created XY.Z folder, remove the various "new in vXY.Z-1"
+# callouts from the copied files.
+#
+# Note: this list may not be exhaustive, see the code below for details.
+#
+# This script attempts to automate these tasks as much as possible.  In
+# addition to copying files around, editing files, etc., it also searches for
+# references to the previous version vX.Y.Z-1 in the newly copied files that
+# now serve as the documentation for vX.Y.Z, and shows those references to the
+# end user (as grep-like output), so they can be edited (or not) as needed.
+#
+# For example, to do all of the admin tasks around bumping the version of
+# CockroachDB from 20.1 to 20.2, and then surfacing the places where additional
+# edits may be needed, run:
+#
+# $ ./bump_major_version.py -d /path/to/docs -f 20.1 -t 20.2
```

Comment:
Might be useful to document here that the script outputs a list of "old_versionish_things" for human consumption?

Reply:
Have added a bit more explanation of that here.  Hope it's clearer? PTAL.

Reply:
Looks good, thank you!